process_report/process_report.py

```diff
@@ -297,23 +348,54 @@ def add_institution(dataframe: pandas.DataFrame):
     return dataframe
 
 
-def export_HU_only(dataframe, output_file):
+def export_billables(dataframe, output_file, invoice_month):
+    invoice_b2_path = f"Invoices/{invoice_month}/NERC {invoice_month}.csv"
```

Comment:
You can figure out the paths in the upload to s3 functions based on the filename and invoice month, this logic doesn't need to be part of the export function, nor invoice-month needs to be part of this function definition.

Reply:
How I wanted to organize the various "export dataframe" functions (like `export_HU_only()`) and `export_invoices()` was that the dataframe functions would return their respective processed dataframe along with the filename for local and S3 storage. This is why I wrote `export_billables()` to return a dict, to allow an association between a data frame and its intended filename. 

I agree that it makes sense to remove the `invoice_month` argument from all the dataframe functions' definition, but I don't think it will make sense to remove the `output_file` argument, since that would make it harder to define the association between each dataframe and their intended local filenames (which may be provided by the user running the script)

Reply:
I agree, I didn't say to remove output file from the function definition. Just invoice month.

Reply:
The current export_* functions from before your change, write the dataframe to file as per output file specified, which is an argument provided to process_csv_report. Those functions currently return nothing. I don't want you to change these functions at all and I don't want you to calculate the s3 path there.
The s3 path calculation you can do in a function called

```python
def upload_to_s3(list_of_filenames, invoice_month):
    invoice_s3_path = f"Invoices/{invoice_month}
    invoice_s3_path_archive = (
        f"Invoices/{invoice_month}/Archive"
    )
    for invoice_file in list_of_filenames:
        # figure out invoice_filename without paths and extension
        path = f"{invoice_s3_path}/{invoice_filename} {invoice_month}.csv"
        archive_path = f{invoice_s3_path_archive}/{filename} {invoice_month} {timestamp}.csv
        invoice_bucket.upload_file(invoice_file, path)
        invoice_bucket.upload_file(invoice_file, archive_path)

upload_to_s3(all the paths of all the written file in a list)
```

You already have the list of output files through the arguments when calling process_csv_report.

Reply:
I've tried my best to follow your feedback, and added a small section in `main()` to add the paths into a list, before passing it into `upload_to_s3()`